DataFormats/PortableTestObjects/src/classes_def.xml

```diff
@@ -6,7 +6,7 @@
     <field name="y_" comment="[nElements_]"/>
     <field name="z_" comment="[nElements_]"/>
     <field name="id_" comment="[nElements_]"/>
-    <field name="metadata()" comment="!"/>
+    <field name="r_" comment="[scalar_]"/>
```

Comment:
This is the only way I found to make ROOT store a single element owned via a pointer.
The scant documentation seems to suggest that `->` should be used for that, but I couldn't get it to work.

Reply:
> This is the only way I found to make ROOT store a single element owned via a pointer.
> The scant documentation seems to suggest that `->` should be used for that, but I couldn't get it to work.

@pcanal, any thoughts?

Reply:
The object pointeed to by a pointer to a class are stored without any annotation.  For pointer to a class, the annotation '->' means that the pointer is guaranteed to never be a nullptr (i.e. the default constructor allocates the object).

For numerical type, it is assumed (because a pointer to a single value is longer than just the value) that it points to an array of numerical elements and indeed, you need to give a size.

Reply:
I see.

So if I wrap the "scalar" type in a `struct` (or `union` ?) I could use the `->` syntax ?

Reply:
With a struct and if you initialize the pointer to non-zero in the default constructor, yes you can use `->` (it wont work for a union).